braid/_braid.c

```diff
@@ -2786,58 +2916,32 @@ _braid_FRefine(braid_Core   core,
       }
    }
 
-   /* Get the next r_fa value to my right */
+   /* Get the next r_ta values to my right.  Note that the r_ta values should be
+    * exchanged in a toroidal fashion in the periodic case. */
    ncomms = 2; /* Upper bound */
    requests = _braid_CTAlloc(MPI_Request, ncomms);
    statuses = _braid_CTAlloc(MPI_Status,  ncomms);
    ncomms = 0;
    if (npoints > 0)
    {
-      braid_Real send_buf[2], recv_buf[2];
-      send_buf[0]=r_fa[0];
-      send_buf[1]=r_ta[0];
-      /* Post r_fa receive */
-      r_fa[npoints] = f_gupper+1;
-      if (iupper < gupper)
+      /* Post r_ta receive */
+      if ((iupper < gupper) || periodic)
       {
-         MPI_Irecv(recv_buf, 2, braid_MPI_REAL, MPI_ANY_SOURCE, 2, comm,
+         MPI_Irecv(&r_ta[r_npoints], 1, braid_MPI_REAL, MPI_ANY_SOURCE, 2, comm,
                    &requests[ncomms++]);
       }
 
-      /* Post r_fa send */
-      if (ilower > 0)
+      /* Post r_ta send (to the left) */
+      if ((ilower > 0) || periodic)
       {
-         _braid_GetBlockDistProc((gupper+1), nprocs, (ilower-1), &prevproc);
-         MPI_Isend(send_buf, 2, braid_MPI_REAL, prevproc, 2, comm,
-                   &requests[ncomms++]);
+         _braid_GetBlockDistProc((gupper+1), nprocs, (ilower-1), periodic, &prevproc);
+         MPI_Isend(&r_ta[0], 1, braid_MPI_REAL, prevproc, 2, comm, &requests[ncomms++]);
```

Comment:
Should we send r_ta[0] or r_ta[-1] ?  I probably have the offsets wrong in my head.

Reply:
This is correct.  I am send my left time value to the processor to my left (which will go in a ghost region on that processor).